golang/pkg/events/events.go

```diff
@@ -1,24 +1,86 @@
 package events
 
 import (
+	"time"
+
+	"github.com/google/uuid"
 	"google.golang.org/protobuf/proto"
 	"google.golang.org/protobuf/types/known/anypb"
+	timestamppb "google.golang.org/protobuf/types/known/timestamppb"
 )
 
-func ToEvent(event proto.Message, source string, id string) ([]byte, error) {
+type cloudEventOptions struct {
+	id     *string
```

Comment:
It looks strange to make them as pointer. I would let them without. You can compare them with empty strings (the `id` and `source`) and using the `IsZero()` for the time (`fields.time.IsZero()`) afterwards, as if they are not initialized the get that value

Reply:
I don't see why se shouldn't use pointer here. It saves a bit of copying around, and the only case that we risk a null pointer value in one of the options is when the user actually passes a `nil` pointer on purpose.

The only argument in favor I see is that we wouldn't need to declare the default variable upfront as you state in your next comment.

Reply:
Well, from my point of view the need to declare the variable to be used looks like a smell, and I wouldn't do it if it's not strictly required. The unique moment where the pointer is really needed is when the `nil` value has some sort of logic (like when you need to marshal/unmarshal strings to this struct).

@CDimonaco what do you think on this? (using pointer or not).